### ISSUE TITLE
Gradle build fails with  "error: package org.apache.http does not exist"

### ISSUE THREAD

**Issue by geekanamika** [systers/malaria-app-android#281]
### What is wrong?
Android 6.0 release removes support for the Apache HTTP client. But this project still uses apache Http.

### Steps to reproduce the issue.
There are two options:
1. use the HttpURLConnection class instead. This API is more efficient because it reduces network use through transparent compression and response caching, and minimizes power consumption
2. ontinue using the Apache HTTP APIs, It'll need to declare `android {
    useLibrary 'org.apache.http.legacy'
}' in build.gradle file. 

Possible solution links : 
* [Stackoverflow](https://stackoverflow.com/questions/31433687/android-gradle-apache-httpclient-does-not-exist)
* [Documentation](https://developer.android.com/about/versions/marshmallow/android-6.0-changes.html#behavior-apache-http-client)

### Expected behaviour.
App should use supported class files. Either it should update to Httpurlconnection or provide library & declare it in build.gradle file.

### Your development environment
Android studio - 3.0.1 stable

### Screenshot or GIF
![screenshot from 2018-03-10 02-07-27](https://user-images.githubusercontent.com/16922933/37228785-cd63401a-2407-11e8-9a14-f6bc389d8f80.png)

### Do you want to work on the issue?
Yes 

_**Note** - Before opening an issue please read [Issue Reporting Best Practices](https://github.com/systers/malaria-app-android/wiki/Issue-Reporting-Best-Practices)._

**Comment by geekanamika** [comment]
@anubhakushwaha @m-murad I would like to work on this issue. Shall i go ahead ?

**Comment by m-murad** [comment]
Go ahead

**Comment by geekanamika** [comment]
@m-murad Which approach should i choose between the two mentioned in "Steps to reproduce". ?

**Comment by saurabhjn76** [comment]
@geekanamika Options 1 using HttpURLConnection seems good to me, let's wait for @m-murad views before proceeding.

**Comment by m-murad** [comment]
@geekanamika @saurabhjn76 
Since we won't be getting a backend for malaria-android anytime soon, I suggest we remove the HTTP stuff altogether and save the user details locally.

If this is not possible I would suggest using the `HttpURLConnection` API

**Comment by m-murad** [comment]
I'm suggesting
Until we get a proper backend we should remove the code related to backend.

In future we can add code according to the backend 
eg If we are interaction with multiple endpoints we can use retorfit.
Else we can use some other API

**Comment by m-murad** [comment]
It's just a suggestion let's see what other people have to say.

**Comment by saurabhjn76** [comment]
yes, this project needs a proper backend, which needs to be discussed first, there are certain better network libraries available which we can add as per our requirements. 
As of now, I believe removing HTTP stuff would involve a lot of unnecessary refactoring which anyhow in future will change, so currently, in my view, we should simply fix this build error, discuss the implementation for the proper backend. Surely let's have insight from more people on it.

**Comment by geekanamika** [comment]
@m-murad @saurabhjn76 Shall i fix this temporarily so that project can be built ? I'll add this discussion in community bonding period in GSoC timline.

**Comment by m-murad** [comment]
How do you propose to fix it? I think that's what the discussion is about 😀

**Comment by m-murad** [comment]
Lets comment code that is using the http stuff. Also add a todo there and mention why it's being commented.

**Comment by geekanamika** [comment]
@m-murad I should have mentioned the step , apologies. 
Exactly the steps you mentioned above. Only commenting the code which involves import error & adding Todo for reason & mentioning issue number for reference.

**Comment by m-murad** [comment]
Great, please send the PR

**Comment by MeepyMay** [comment]
The backend to most Web applications with Peace Corps is the MACC project.

**Comment by m-murad** [comment]
Did maac used to save the details from malaria application?

**Comment by geekanamika** [comment]
@MeepyMay @m-murad Server address is "http://pc-web-dev.systers.org/api/malaria_users" in application.
